### PR TITLE
Feature/sign up flow

### PR DESCRIPTION
**Pull request by kamran-khalid-shah**
Sign up, sign in, log out flow fixed

### CODE REVIEW

File: app/controllers/user_companies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      user = User.unscope(where: :company_id).find_by! email: params[:email]
+      company = Company.find_by! id: user.company_id
+    rescue Exception
+      success = false
+    end
+    if success
+      redirect_to new_user_session_url(subdomain: company.subdomain) and return
+    else
+      if params[:email].length.zero?
+        flash.now[:error] = 'Email cannot be empty!'
```
Comment (kamran-khalid-shah) [inline app/controllers/user_companies_controller.rb]: i18n

File: app/controllers/user_companies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      else
+        flash.now[:error] = 'Invalid email entered!'
```
Comment (kamran-khalid-shah) [inline app/controllers/user_companies_controller.rb]: i18n

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -4,6 +4,10 @@
 class ApplicationController < ActionController::Base
   around_action :scope_current_company
 
+  def after_sign_out_path_for(resource)
```
Comment (bk-az) [inline app/controllers/application_controller.rb:7]: private method

Comment (bk-az) [inline app/controllers/application_controller.rb:7]: move to session controller

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -4,6 +4,10 @@
...
+    new_user_session_url(subdomain: request.subdomain)
```
Comment (bk-az) [inline app/controllers/application_controller.rb]: no need to specify subdomain

File: app/controllers/user_companies_controller.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      format.html
+    end
+  end
+
+  def search_by_email
+    success = true
+    begin
+      user = User.unscope(where: :company_id).find_by! email: params[:email]
+      company = Company.find_by! id: user.company_id
+    rescue Exception
```
Comment (bk-az) [inline app/controllers/user_companies_controller.rb]: rescue ActiveRecord::RecordNotFound

File: app/controllers/user_companies_controller.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      success = false
+    end
+    if success
+      redirect_to new_user_session_url(subdomain: company.subdomain) and return
+    else
+      if params[:email].length.zero?
```
Comment (bk-az) [inline app/controllers/user_companies_controller.rb]: params[:email].blank?

File: app/views/home/_header.html.erb [hunk 7]
```diff
@@ -27,9 +27,13 @@
         <a class="nav-link disabled" href="#">Disabled</a>
       </li>
     </ul>
-    <form class="form-inline my-2 my-lg-3 float-right">
-      <input class="form-control mr-sm-2 float-right" type="search" placeholder="Search" aria-label="Search">
-      <button class="btn btn-outline-success my-2 my-sm-0 float-right" type="submit">Search</button>
-    </form>
+    
+    <% if current_user == nil %>
```
Comment (bk-az) [inline app/views/home/_header.html.erb]: if current_user.blank?

File: app/views/user_companies/find.html.erb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<h2>Enter Email</h2>
+
+<%= render "shared/flash" %>
```
Comment (bk-az) [inline app/views/user_companies/find.html.erb]: move this to application.html.erb file (layout)

File: app/views/shared/_flash.html.erb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<% flash.each do |key, value| %>
```
Comment (bk-az) [inline app/views/shared/_flash.html.erb:1]: <% flash.each do |key, value| %>
      <div class="<%= flash_class(key) %>" role="alert">
        <%= value %>
        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
    <% end %>

Comment (bk-az) [reply app/views/shared/_flash.html.erb:1]: We should also consider case when value is an array

    <% flash.each do |key, value| %>
      <div class="<%= flash_class(key) %>" role="alert">
        <a class="close" data-dismiss="alert" href="#_">&times;</a>
        <% if value.is_a?(Array) %>
          <ul>
            <% value.each do |val| %>
              <li><%= val %></li>
            <% end %>
          </ul>
        <% else %>
          <%= value %>
        <% end %>
      </div>
    <% end %>

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,10 @@
 module ApplicationHelper
+  def flash_class(level)
```
Comment (bk-az) [inline app/helpers/application_helper.rb:2]: def flash_class(level)
      case level.to_sym
      when :notice then 'alert alert-info'
      when :success then 'alert alert-success'
      when :error then 'alert alert-danger'
      when :alert then 'alert alert-warning'
      end
    end

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -11,6 +11,7 @@
 
   <body>
     <%= render :partial => "home/header" %>
+    <%= render :partial => "shared/flash" %>
```
Comment (bk-az) [inline app/views/layouts/application.html.erb]: <div id="flash_messages">
      <%= render :partial => "shared/flash" %>
    </div>

File: app/models/user.rb [hunk 12]
```diff
@@ -26,4 +26,10 @@ class User < ApplicationRecord
   has_many :watching_tasks, through: :tasks_watchers, source: :task
   has_many :projects_users, dependent: :destroy
   has_many :projects, through: :projects_users
+
+  protected
+
+  def confirmation_required?
+    false
```
Comment (bk-az) [inline app/models/user.rb]: `created_at < 7.days.ago`

**Pull request merged**